jscomp/others/js_option.ml

```diff
@@ -59,5 +59,27 @@ let andThen f x =
   | None -> None 
   | Some x -> f x [@bs]
 
+let map f x =
+  match x with
+  | None -> None
+  | Some x -> Some (f x [@bs])
 
+let orValue a x =
+  match x with
+  | None -> a
+  | Some x -> x
```

Comment:
> @OvermindDL1 I thought I'd do that, but they said on Discord that this naming was intended to make sense to javascript developers. I questioned on whether we should instead just use the standard functions that you posted as that would make the code more reusable between server and client applications.

Considering I'm planning to have bucklescript-tea have a server-side part soon it better work in native code as well.  Even from what little of the horror's I've seen of the 'JS perspective' so far I still do not see how some of their names make any sense, like `andThen` for a concept as such as in here...  >.>